db/multi.go

```diff
@@ -50,39 +66,50 @@ func (mi *MultiInserter) Add(row []interface{}) error {
 // for gorp to use in place of the query's question marks. Currently only
 // used by .Insert(), below.
 func (mi *MultiInserter) query() (string, []interface{}) {
-	questionsRow := strings.TrimRight(strings.Repeat("?,", mi.numFields), ",")
+	questionsRow := strings.TrimRight(strings.Repeat("?,", len(mi.fields)), ",")
 
 	var questionsBuf strings.Builder
 	var queryArgs []interface{}
 	for _, row := range mi.values {
+		// Safety: We are interpolating a string that will be used in a SQL
+		// query, but we constructed that string in this function and know it
+		// consists only of question marks joined with commas.
 		fmt.Fprintf(&questionsBuf, "(%s),", questionsRow)
 		queryArgs = append(queryArgs, row...)
 	}
 
 	questions := strings.TrimRight(questionsBuf.String(), ",")
 
+	// Safety: we are interpolating `mi.returningColumn` into an SQL query. We
+	// know it is a valid unquoted identifier in MariaDB because we verified
+	// that in the constructor.
```

Comment:
Yep, "Constructor" is a generic term used across many languages for "function that returns a Thing". _Most_ of the time, a constructor returns a "pointer to a Thing" because, if the object is complex enough for a constructor function to be worthwhile, then it's _probably_ also complex enough to be worth allocating on the heap and referencing via a pointer.